pkg/diff/util.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"math/rand"
+
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/types"
+)
+
+func equalStrings(str1, str2 []string) bool {
+	if len(str1) != len(str2) {
+		return false
+	}
+	for i := 0; i < len(str1); i++ {
+		if str1[i] != str2[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func equalTableInfo(tableInfo1, tableInfo2 *model.TableInfo) (bool, error) {
+	// check columns
+	if len(tableInfo1.Columns) != len(tableInfo2.Columns) {
+		return false, nil
+	}
+
+	for j, col := range tableInfo1.Columns {
+		if col.Name.O != tableInfo2.Columns[j].Name.O {
+			return false, nil
+		}
+		if col.Tp != tableInfo2.Columns[j].Tp {
+			return false, nil
+		}
+	}
+
+	// check index
+	if len(tableInfo1.Indices) != len(tableInfo2.Indices) {
+		return false, nil
+	}
+
+	for i, index := range tableInfo1.Indices {
+		index2 := tableInfo2.Indices[i]
+		if index.Name.O != index2.Name.O {
+			return false, nil
+		}
+		if len(index.Columns) != len(index2.Columns) {
+			return false, nil
+		}
+		for j, col := range index.Columns {
+			if col.Name.O != index2.Columns[j].Name.O {
+				return false, nil
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func ignoreColumns(tableInfo *model.TableInfo, columns []string) *model.TableInfo {
```

Comment:
I don't think we need to change `tableinfo`,  e.g. a table with a unique key (id, name,age), if we remove id, name and age maybe not unique, how to sort them?